components/ThemeWrapper.svelte

```diff
@@ -11,13 +11,27 @@
   import presets from './presets'
   import setCSS from '../support/setCSS'
   import toggleTheme from '../support/toggleTheme'
-  import { currentTheme } from '../support/store'
+  import toggleMode from '../support/toggleMode'
+  import { currentMode, currentTheme } from '../support/store'
+
+  // detect dark mode
+  const darkModeQuery = window.matchMedia('(prefers-color-scheme: dark)');
 
   /**
    * Specify the key used for local storage
    * @type {string} [key='__svelte-themer__theme']
    */
   export let key = STORAGE_KEY
+  /**
+   * Dark mode localStorage key
+   * @type {string}
+   */
+  const darkModeLSKey = `${key}__mode`;
+  /**
+   * Themes
+   * @type {string} themes - name of the theme to use
+   */
+  export let theme = null
```

Comment:
Initial `theme` can be defined here by name along with the `mode`. if either one of these values exist they override everything else. 

Current flow:
> user-defined -> localStorage -> fallback value

Example
```jsx
<ThemeWrapper theme="forest">

<ThemeWrapper theme="forest" mode="dark">
```

Reply:
If I understand correctly in the end we're not expecting the `theme` choice to change, correct? This setup is growing on me, I'll admit. And for the `mode` I think that'd be a good spot to have the options be one of `light | dark | prefers`. I'm fine with the theme prop just want to be sure we're not creating the CSS for the other presets.

Reply:
for the dark mode storage I'd like to keep it to just the one key. Perhaps we could create the value as `theme--mode`

Reply:
> If I understand correctly in the end we're not expecting the theme choice to change, correct? This setup is growing on me, I'll admit. 

Yes and no, I guess. It wouldn't matter if it were to change in the end. This just allows the developer two ways of defining the initial theme 

1. Pass an ordered `themes` array so the theme you want is first, and do not set the `theme` prop.
2. Pass an unordered `themes` array and be explicit by setting the `theme` prop.
3. Do both.

> And for the mode I think that'd be a good spot to have the options be one of light | dark | prefers. I'm fine with the theme prop just want to be sure we're not creating the CSS for the other presets.

This change/approach is where we differ, and it doesn't work for what you're after since you want to treat dark mode and light mode as independent themes. In order to get what you want, you could add another prop `dark-mode-theme`||`theme-dark`(?) where the user could provide the theme name to use when `preferred-color-scheme` is `dark`, since the non-dark theme should be considered the `light` theme (the browsers treat the default or no pref as `light`).

If the above were introduced, the `mode` prop could support these values:

1. `dark` -> when set uses the theme defined inside `dark-mode-theme` or `theme-dark`  
2. `light` -> when set uses the theme defined inside  `theme` or the first theme in the themes array if `theme` is undefined. 
3. 'auto' -> check for dark mode `window.matchMedia('(prefers-color-scheme: dark)')` and set light or dark based on `matchMedia` result.

> for the dark mode storage I'd like to keep it to just the one key. Perhaps we could create the value as theme--mode 

I originally had, and preferred, `theme--mode`, but thought you might want to keep things consistent with the LS key so I threw it in there. 👍